### PR TITLE
Andrew/profile image

### PR DESCRIPTION
**Pull request by amillman**
simulator screen shot feb 6 2016 1 19 05 am

### CODE REVIEW

File: sage-ios/SAGE/SAGE/ProfileImageView.swift [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    // MARK: - Public Methods
+    //
+    func setDiameter(diameter: CGFloat) {
+        self.setSize(width: diameter, height: diameter)
+        self.imageView.setSize(width: diameter, height: diameter)
+        self.imageView.layer.cornerRadius = diameter / 2
+        
+        let badgeBorderSize = diameter/badgeSizeFactor
+        self.badgeBorder.setSize(width: badgeBorderSize, height: badgeBorderSize)
+        self.badgeBorder.layer.cornerRadius = badgeBorderSize / 2
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/ProfileImageView.swift]: you do this already in layout subviews!

File: sage-ios/SAGE/SAGE/ProfileImageView.swift [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        self.badgeView.centerInSuperview()
+    }
+    
+    //
+    // MARK: - Public Methods
+    //
+    func setDiameter(diameter: CGFloat) {
+        self.setSize(width: diameter, height: diameter)
+        self.imageView.setSize(width: diameter, height: diameter)
+        self.imageView.layer.cornerRadius = diameter / 2
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/ProfileImageView.swift:81]: to be consistent, all the layer should either be in this function or in layout subviews

File: sage-ios/SAGE/SAGE/ProfileImageView.swift [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        
+        let badgeBorderSize = diameter/badgeSizeFactor
+        self.badgeBorder.setSize(width: badgeBorderSize, height: badgeBorderSize)
+        self.badgeBorder.layer.cornerRadius = badgeBorderSize / 2
+        
+        let badgeViewSize = CGFloat(Int(badgeBorderSize * badgeViewPercentage))
+        self.badgeView.setSize(width: badgeViewSize, height: badgeViewSize)
+        self.badgeView.layer.cornerRadius = badgeViewSize / 2
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/ProfileImageView.swift]: ya this too

File: sage-ios/SAGE/SAGE/ProfileImageView.swift [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        default:
+            self.badgeBorder.alpha = 0
+        }
+        if !showBadge {
+            self.badgeBorder.alpha = 0
+        }
+    }
+    
+    func cancelImageRequestOperation() {
+        self.imageView.cancelImageRequestOperation()
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/ProfileImageView.swift]: should this be in deinit too

**Comment by sameerav** [comment]
looks fine when i test it so just that small thing!

File: sage-ios/SAGE/SAGE/ProfileImageView.swift (hunk 4 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/ProfileImageView.swift]: good catch

**Pull request merged**